src/lib/import-export/types.ts

```diff
@@ -0,0 +1,29 @@
+import { EventEmitter } from 'events';
+import { ExportEventType } from './export/events';
+import { ImportEventType } from './import/events';
+
+export interface ImportExportEventData {
+	event: ImportEventType | ExportEventType;
+	data: unknown;
+}
+
+export const handleEvents = <
+	T extends Record< string, string >,
+	K extends ImportEventType | ExportEventType,
+>(
+	emitter: Partial< EventEmitter >,
+	onEvent: ( data: ImportExportEventData ) => void,
+	events: T
+) => {
+	Object.values( events ).forEach( ( eventName ) => {
+		if ( ! emitter.on ) {
+			return;
+		}
+		emitter.on( eventName, ( data: unknown ) => {
```

Comment:
Probably it's not necessary to remove the listeners, but it would be great to clean them up to avoid potential undesired side effects and memory leaks.

Reply:
I created a PR to update this: https://github.com/Automattic/studio/pull/383